firefox-webext/background.js

```diff
@@ -17,3 +106,59 @@ chrome.browserAction.onClicked.addListener(function(tab) {
     });
   });
 });
+
+function storeIssues() {
+  chrome.storage.local.set({issuesList, tsRefresh}, function(){
+    if (chrome.runtime.lastError) {
+      console.log(chrome.runtime.lastError);
+    } else {
+      console.log("Local store complete.");
+    }
+  });
+}
+
+function storageCheck(item){
+  if (chrome.runtime.lastError) {
+    console.log(chrome.runtime.lastError);
+  } else {
+    if (Object.keys(item).length === 0 && item.constructor === Object){
+      getIssues(); // if nothing in storage fetch the list from github
+    } else {
+      issuesList = item.issuesList;
+      tsRefresh = item.tsRefresh;
+      console.log('Issues loaded from storage.');
+      checkRefresh();
+    }
+  }
+}
+
+function extractDomain(url) {
+   var domain;
+   if (url.indexOf("://") > -1) {
+     domain = url.split('/')[2];
+   }
+   else {
+     domain = url.split('/')[0];
+   }
+   domain = domain.split(':')[0];
+   return domain;
+}
+
+function navigationChange(details) {
+  var url = extractDomain(details.url);
+  var tab = details.tabId;
+  var parentFrameId = details.parentFrameId;
+  if (parentFrameId == -1){
```

Comment:
webNavigation.onCompleted gets fired a lot when pages are loading. If parentFrameID is -1 it's the current URL of that tab and the one we're interested in.